web/src/app/services/ServiceLabelEditDialog.js

```diff
@@ -26,56 +26,63 @@ const query = gql`
   }
 `
 
-export default class ServiceLabelCreateDialog extends React.PureComponent {
-  static propTypes = {
-    serviceID: p.string.isRequired,
-    labelKey: p.string.isRequired,
-    onClose: p.func,
-  }
-
-  state = {
-    value: null,
-    errors: [],
-  }
-
-  render() {
-    return this.renderQuery()
-  }
+export default function ServiceLabelCreateDialog(props) {
+  const [value, setValue] = useState(null)
+  const { onClose, labelKey, serviceID } = props
 
-  renderQuery() {
+  function renderDialog(data, commit, status) {
+    const { loading, error } = status
     return (
-      <Query
-        noPoll
-        query={query}
-        variables={{ serviceID: this.props.serviceID }}
-        render={({ data }) =>
-          this.renderMutation(
-            data.service.labels.find((l) => l.key === this.props.labelKey),
-          )
+      <FormDialog
+        title='Update Label Value'
+        loading={loading}
+        errors={nonFieldErrors(error)}
+        onClose={onClose}
+        onSubmit={() => {
+          if (!value) {
+            return onClose()
+          }
+          return commit({
+            variables: {
+              input: {
+                ...value,
+                target: { type: 'service', id: serviceID },
+              },
+            },
+          })
+        }}
+        form={
+          <ServiceLabelForm
+            errors={fieldErrors(error)}
+            editValueOnly
+            disabled={loading}
+            value={value || { key: data.key, value: data.value }}
+            onChange={(value) => setValue({ value })}
+          />
         }
       />
     )
   }
 
-  renderMutation(data) {
+  function renderMutation(data) {
     return (
       <Mutation
         mutation={mutation}
-        onCompleted={this.props.onClose}
+        onCompleted={onClose}
         update={(cache) => {
           const { service } = cache.readQuery({
             query,
-            variables: { serviceID: this.props.serviceID },
+            variables: { serviceID: serviceID },
           })
           const labels = (service.labels || []).filter(
-            (l) => l.key !== this.state.value.key,
+            (l) => l.key !== value.key,
           )
-          if (this.state.value.value) {
-            labels.push({ ...this.state.value, __typename: 'Label' })
+          if (value.value) {
+            labels.push({ ...value, __typename: 'Label' })
           }
           cache.writeQuery({
             query,
-            variables: { serviceID: this.props.serviceID },
+            variables: { serviceID: serviceID },
```

Comment:
since the names are the same, in Javascript you can reference `serviceID` once:
```suggestion
            variables: { serviceID },
```